extensions/wikia/Search/classes/FandomSearch.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Wikia\Search;
+
+use \Wikia\Logger\WikiaLogger;
+
+class FandomSearch {
+	const FANDOM_API = 'http://fandom.wikia.com/wp-json/wp/v2/';
+	const FANDOM_API_POSTS_ENDPOINT = 'posts';
+	const FANDOM_SEARCH_PAGE = 'http://fandom.wikia.com/?s=';
+	const RESULTS_COUNT = 5;
+	const VERTICALS_PARENT_CATEGORY_ID = 3;
+	const CATEGORY_TAXONOMY_NAME = 'category';
+	const CACHE_TTL = 12 * 60 * 60;
+
+	public static function getStoriesWithCache( $query ) {
+		return \WikiaDataAccess::cache(
+			wfSharedMemcKey( 'FandomSearch', $query ),
+			static::CACHE_TTL,
+			function() use ( $query ) {
+				return static::getStories( $query );
+			}
+		);
+	}
+
+	public static function getStories( $query ) {
+		$url = static::buildUrl( $query );
+		$method = 'GET';
+		$options = [
+			'curlOptions' => [
+				// TODO set when we use internal copy
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this "TODO" comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1135)